README.md

```diff
@@ -20,127 +19,273 @@ let thirdNumber = numbers[2]
 ]
- 
+
  if !moonsOfJupiter.isEmpty {
     for moon in moonsOfJupiter {
         print("\(moon) is a moon orbiting Jupiter.")
     }
  }
- 
+
  // Europa is a moon orbiting Jupiter.
  // Ganymede is a moon orbiting Jupiter.
  // Io is a moon orbiting Jupiter.
  // Callisto is a moon orbiting Jupiter.
 ```
 
+--------------------------------------------------------------------------------
+
+In the last lesson you briefly saw a 'method', which can be thought of as a pre-written function that helps you accomplish common tasks. They are contextual, in that, for example, a method that works on an array may not work for a String.
+
+The method you saw in the last example was `.append()` that adds a value to the end of an array. You could achieve this in several ways which would involve far more code and calculation, I think you'll agree that `.append()` is far more convenient. There are many other methods for arrays, and in this lesson you'll look at some others.
+
+## Enumeration
+
+Before going any further, there's one other array concept to introduce.
+
+Say you have a list of instruction steps, if steps need to be added, removed, or changed at a particular index in the array, then you need to know what the index of the value is. When you have a simple list of unchanging values, then maybe this is simple, but with a larger list processed in a loop, there is another way, you can use the `.enumerate` method to have not only the value of the current loop index, but the index position itself.
+
+```swift
+var toMakeTea = [
+    "Boil Water",
+    "Add tea bag to cup",
+    "Wait ten minutes",
+    "Add Milk",
+    "Drink"
+]
+
+for (index, step) in toMakeTea.enumerate() {
+    print("\(index + 1). \(step)")
+}
+
+// 1. Boil Water
+// 2. Add tea bag to cup
+// 3. Wait ten minutes
+// 4. Add Milk
+// 5. Drink
+```
+
+In this example you first create an array of the steps to make tea (and yes, there are intentional mistakes!) and add a couple of extra things to the `for in` loop. First inside the brackets you add variables to hold the index and value variables, the names of these are up to you. Using `index` is of course a useful name, but you can call it whatever you want. At the end of the array variable you want to iterate through, add the `.enumerate()` method. Now inside the loop you have access to the value in the current iteration as well as it's index in the array. Above, you add '1' to the index value as unlike computers, humans don't count from 0!
+
+## Check an array has values
+
+Before undertaking any further actions, it's a good idea to check the array you would like to work with actually has any values in it. Of course there's a method to help, `.isEmpty`.
+
+```swift
+if toMakeTea.isEmpty {
+   print("No tea yet :(")
+} else {
+   print("Ready to make tea! :)")
+}
+```
+
+This method checks to see if the array contains any values, but the array still needs to actually exist (i.e. it's you have initiated it) in the first place.
+
```

Comment:
I like the idea of showing them an example of what you're describing here. Like this:

![Hello](http://i.imgur.com/jrailM6.png)

Asking the student to locate the problem. The problem being that we've declared that the variable named `lionelRichieSongs` will be of type [`String`], an array that contains `String` values but we didn't give it an initial value (initialized) yet. If there's no starting place to work from, we will be unable to call any methods on it.